src/xtopcom/xvledger/src/xvblock.cpp

```diff
@@ -1065,14 +1076,27 @@ namespace top
             }
             
             const std::string data_to_sign((const char*)stream.data(),stream.size());
-            return hash(data_to_sign);
+            //if m_modified_count != 0, rebuild hash 
+            if (check_modified_change() > 0) {
+                reset_modified_count();
+                m_sign_hash = hash(data_to_sign);
+                return m_sign_hash;
+            }else{
+                return  hash(data_to_sign);
+            }
         }
         
         const std::string  xvqcert_t::build_block_hash()    //recalculate the hash of block, note: block_hash is equal as cert'hash
         {
-            std::string  cert_bin_data;
-            serialize_to_string(cert_bin_data);
-            return hash(cert_bin_data);
+            if(check_modified_change() > 0 || m_cert_hash.empty() == true ){
```

Comment:
for check_modified_change, return boolean may be better

Reply:
fix it.